src/ctap/pin_protocol_v1.rs

```diff
@@ -0,0 +1,1137 @@
+        let mut pin_token = vec![];
+        for item in blocks.iter().take(PIN_TOKEN_LENGTH / 16) {
+            pin_token.extend(item);
+        }
+
+        #[cfg(feature = "with_ctap2_1")]
+        {
+            self.permissions = 0x03;
+            self.permissions_rp_id = None;
+        }
+
+        Ok(AuthenticatorClientPinResponse {
+            key_agreement: None,
+            pin_token: Some(pin_token),
+            retries: None,
+        })
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    fn process_get_pin_uv_auth_token_using_uv_with_permissions(
+        &self,
+        _: CoseKey,
+    ) -> Result<AuthenticatorClientPinResponse, Ctap2StatusCode> {
+        // User verifications is only supported through PIN currently.
+        #[cfg(not(feature = "with_ctap2_1"))]
+        {
+            Err(Ctap2StatusCode::CTAP1_ERR_INVALID_COMMAND)
+        }
+        #[cfg(feature = "with_ctap2_1")]
+        {
+            Err(Ctap2StatusCode::CTAP2_ERR_INVALID_SUBCOMMAND)
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    fn process_get_uv_retries(&self) -> Result<AuthenticatorClientPinResponse, Ctap2StatusCode> {
+        // User verifications is only supported through PIN currently.
+        #[cfg(not(feature = "with_ctap2_1"))]
+        {
+            Err(Ctap2StatusCode::CTAP1_ERR_INVALID_COMMAND)
+        }
+        #[cfg(feature = "with_ctap2_1")]
+        {
+            Err(Ctap2StatusCode::CTAP2_ERR_INVALID_SUBCOMMAND)
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    fn process_set_min_pin_length(
+        &mut self,
+        persistent_store: &mut PersistentStore,
+        min_pin_length: u8,
+        min_pin_length_rp_ids: Option<Vec<String>>,
+        pin_auth: Option<Vec<u8>>,
+    ) -> Result<(), Ctap2StatusCode> {
+        if min_pin_length_rp_ids.is_some() {
+            return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_EXTENSION);
+        }
+        if persistent_store.pin_hash().is_some() {
+            match pin_auth {
+                Some(pin_auth) => {
+                    // TODO(kaczmarczyck) not mentioned, but maybe useful?
+                    // if persistent_store.pin_retries() == 0 {
+                    //     return Err(Ctap2StatusCode::CTAP2_ERR_PIN_BLOCKED);
+                    // }
+                    if self.consecutive_pin_mismatches >= 3 {
+                        return Err(Ctap2StatusCode::CTAP2_ERR_PIN_AUTH_BLOCKED);
+                    }
+                    let mut message = vec![0xFF; 32];
+                    message.extend(&[0x06, 0x08]);
+                    message.extend(&[min_pin_length as u8, 0x00, 0x00, 0x00]);
+                    // TODO(kaczmarczyck) commented code is useful for the extension
+                    // if !cbor::write(cbor_array_vec!(min_pin_length_rp_ids), &mut message) {
+                    //     return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_RESPONSE_CANNOT_WRITE_CBOR);
+                    // }
```

Comment:
When do you expect to resolve these TODO? Linking to a tracking GitHub issue would be better.

Reply:
Done.